pkg/controller/controllercmd/builder.go

```diff
@@ -134,6 +138,23 @@ func (b *ControllerBuilder) WithLeaderElection(leaderElection configv1.LeaderEle
 	return b
 }
 
+// WithVersion accepts a getting that provide binary version information that is used to report build_info information to prometheus
+func (b *ControllerBuilder) WithVersion(versionGetter func() version.Info) *ControllerBuilder {
+	buildInfo := metrics.NewGaugeVec(
+		&metrics.GaugeOpts{
+			Name: strings.Replace(b.componentNamespace, "-", "_", -1) + "_build_info",
```

Comment:
you want to do this part during the actual .Run so you're guaranteed this componentNamespace is set.